src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -110,23 +116,36 @@ TableOfContentsInputItem GetItem(HeadingBlock block)
                 }
 
                 var link = block.Inline.FirstOrDefault(l => l is LinkInline);
-                var literalLink = block.Inline.FirstOrDefault(l => l is LiteralInline);
                 if (link != null && link is LinkInline linkInline)
                 {
                     if (!string.IsNullOrEmpty(linkInline.Url))
                         currentItem.Href = linkInline.Url;
                     if (!string.IsNullOrEmpty(linkInline.Title))
                         currentItem.DisplayName = linkInline.Title;
 
-                    literalLink = linkInline.FirstOrDefault(l => l is LiteralInline);
+                    currentItem.Name = GetName(linkInline.Select(l => l as LeafInline));
                 }
 
-                if (literalLink != null && literalLink is LiteralInline literalInline)
-                {
-                    currentItem.Name = literalInline.Content.ToString();
-                }
+                currentItem.Name = currentItem.Name ?? GetName(block.Inline.Select(l => l as LeafInline));
 
                 return currentItem;
+
+                string GetName(IEnumerable<LeafInline> literalInlines)
```

Comment:
Is `IEnumerable<Inline>` sufficient. I see `ContainerInline` has several derived types like `DelimiterInline` and `EmphasisInline`, will it affect the behavior here?

Reply:
yes, check my first changed test case